packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -137,14 +171,14 @@ class ScrollController extends ChangeNotifier {
     super.dispose();
   }
 
-  static ScrollPosition createDefaultScrollPosition(ScrollPhysics physics, ScrollContext context, ScrollPosition oldPosition) {
-    return new ScrollPositionWithSingleContext(
-      physics: physics,
-      context: context,
-      oldPosition: oldPosition,
-    );
-  }
-
+  /// Creates a [ScrollPosition] for use by a controlled [Scrollable] widget.
+  ///
+  /// Subclasses can override this function to customize the [ScrollPosition]
+  /// used by the scrollable widgest they control. For example, [PageController]
+  /// overrides this function to return a page-oriented scroll position
+  /// subclass.
+  ///
+  /// By default, returns a [ScrollPositionWithSingleContext].
   ScrollPosition createScrollPosition(
```

Comment:
You could try and explain a little more about what would motivate someone to create a custom scroll position. For example it might be worth explaining that PageView uses one to keep the selected page scrolled to the center when the scrollable's viewport size changes.